pdns/syncres.cc

```diff
@@ -3363,16 +3363,21 @@ RCode::rcodes_ SyncRes::updateCacheFromRecords(unsigned int depth, LWResult& lwr
       isAA = false;
       expectSignature = false;
     }
+    else if (isDNAMEAnswer && (i->first.place != DNSResourceRecord::ANSWER || i->first.type != QType::DNAME || !qname.isPartOf(i->first.name))) {
```

Comment:
I believe they will remain exclusive since we don't process the CNAME when we get a DNAME, and re-generate the CNAME ourselves instead. But even if they don't, do we really care about evaluating the second condition when the first is true, given that the outcome is the same?

Reply:
right.